plugins/package-json-hook/src/package-json-helper.ts

```diff
@@ -44,36 +40,57 @@ export default abstract class PackageJson extends Hook<PackageJsonState> {
     return this._packageJson
   }
 
-  private get hookPath(): string[] {
-    return Array.isArray(this.field) ? [...this.field, this.key] : [this.field, this.key]
-  }
-
   async check(): Promise<boolean> {
     const packageJson = await this.getPackageJson()
-    return get(packageJson, this.hookPath)?.includes(this.hook)
+    return Object.entries(this.options).every(([field, object]) =>
```

Comment:
**suggestion:** maybe this could be refactored to a be a little clearer regarding the logic behind the steps

Reply:
done, hopefully clearer!